pennylane/operation.py

```diff
@@ -756,6 +756,7 @@ def parameters(self):
     @property
     def hyperparameters(self):
         """dict: Dictionary of non-trainable variables that define this operation."""
+        # pylint: disable=attribute-defined-outside-init
         if hasattr(self, "_hyperparameters"):
             return self._hyperparameters
         self._hyperparameters = {}
```

Comment:
Is there a better way which does not access the private attribute? Maybe a setter?

Reply:
Could we just define `self._hyperparameters` in `__init__`?